// vuepress-plugin-smplayer/components/Aplayer/AplayerComponent.js
import Aplayer from "./Aplayer";
import Vue from "vue";
import merge from "ts-deepmerge";
export default Vue.extend({
    props: {
        src: {
            type: Object,
            required: true,
        },
        on: {
            type: Object,
            default: () => APLAYER.on,
            required: false,
        },
    },
    render() {
        return (<div class="smplayer">
        <div ref="sbplayer"/>
      </div>);
    },
    data() {
        return {
            player: {},
        };
    },
    async mounted() {
        let on = merge(APLAYER.on, this.on);
        let src = {
            ...merge(APLAYER.src, this.src),
            container: this.$refs.sbplayer,
        };
        this.player = new Aplayer(src);
        await this.player.InitPlayer();
        this.player.AddOnEvent(on);
        this.$nextTick(() => {
            if (this.src.fixed) {
                const app = document.querySelector("#app");
                app?.append(this.$el);
            }
        });
    },
    beforeDestroy() {
        this.player?.DestroyPlayer();
    },
});
