// vuepress-plugin-smplayer/components/Artplayer/ArtplayerComponent.js
import Artplayer from "./Artplayer";
import Vue from "vue";
import merge from "ts-deepmerge";
export default Vue.extend({
    props: {
        src: {
            type: Object,
            required: true,
        },
        on: {
            type: Object,
            default: () => ARTPLAYER.on,
            required: false,
        },
        width: {
            type: String,
            default: ARTPLAYER.width,
            required: false,
        },
        height: {
            type: Array,
            default: () => ARTPLAYER.height,
            required: false,
        },
    },
    render() {
        return (<div class="smplayer">
        <div ref="sbplayer" style={`width: ${this.width}`}/>
      </div>);
    },
    data() {
        return {
            player: {},
        };
    },
    async mounted() {
        let sbplayer = this.$refs.sbplayer;
        let on = merge(ARTPLAYER.on, this.on);
        let src = {
            ...merge(ARTPLAYER.src, this.src),
            container: sbplayer,
        };
        this.player = new Artplayer(src);
        await this.player.InitPlayer();
        this.player.AddOnEvent(on);
        sbplayer.style.height =
            sbplayer.scrollWidth * this.height[0] + this.height[1] + "px";
    },
    beforeDestroy() {
        this.player?.DestroyPlayer();
    },
});
