// vuepress-plugin-smplayer/components/Dplayer/DplayerComponent.js
import Dplayer from "./Dplayer";
import Vue from "vue";
import merge from "ts-deepmerge";
export default Vue.extend({
    props: {
        src: {
            type: Object,
            required: true,
        },
        on: {
            type: Object,
            default: () => DPLAYER.on,
            required: false,
        },
        width: {
            type: String,
            default: DPLAYER.width,
            required: false,
        },
        height: {
            type: Array,
            default: () => DPLAYER.height,
            required: false,
        },
    },
    render() {
        return (<div class="smplayer">
        <div ref="sbplayer" style={`width: ${this.width}`}/>
      </div>);
    },
    data() {
        return {
            player: {},
        };
    },
    async mounted() {
        let on = merge(DPLAYER.on, this.on);
        let src = {
            ...merge(DPLAYER.src, this.src),
            container: this.$refs.sbplayer,
        };
        this.player = new Dplayer(src);
        await this.player.InitPlayer();
        this.player.AddOnEvent(on);
    },
    beforeDestroy() {
        this.player?.DestroyPlayer();
    },
});
